http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/net/InputShutdownCheckingSslSocket.java

```diff
@@ -65,17 +163,26 @@ public void write(byte[] b, int off, int len) throws IOException {
         }
 
         private void checkInputShutdown() throws IOException {
-            if (sock.isInputShutdown()) {
+            if (sslSocket.isInputShutdown()) {
                 throw new IOException("Remote end is closed.");
             }
 
             try {
-                sock.getInputStream();
+                if (dataAvailableOnLayeredSocket()) {
+                    LOG.debug(() -> String.format("%d bytes available in layered socket, calling SSLSocket's read() to try and "
+                                                 + "consume the TLS record(s) if present."));
```

Comment:
%d was not specified.

Reply:
Oops! That's probably the build failures 😅